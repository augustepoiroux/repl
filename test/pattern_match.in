{"cmd": "def foo : (fun ((x, _, _) : Nat × Nat × Nat) => x = x) = fun _ => True := by sorry"}

{"tactic": "funext ⟨x, y, z⟩", "proofState": 0}

{"tactic": "show (x = x) = True", "proofState": 1}

{"tactic": "rw [eq_self_iff_true]", "proofState": 2}

{"tactic": "simp", "proofState": 0}

{"cmd": "def foo : (fun ((x, _, _) : Nat × Nat × Nat) => x = x) = fun _ => True := by simp"}
