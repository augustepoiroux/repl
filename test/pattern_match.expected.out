{"sorries":
 [{"proofState": 0,
   "pos": {"line": 1, "column": 77},
   "goal":
   "⊢ (fun x =>\n      match x with\n      | (x, fst, snd) => x = x) =\n    fun x => True",
   "endPos": {"line": 1, "column": 82}}],
 "messages":
 [{"severity": "warning",
   "pos": {"line": 1, "column": 4},
   "endPos": {"line": 1, "column": 7},
   "data": "declaration uses 'sorry'"}],
 "env": 0}

{"proofStatus": "Incomplete: open goals remain",
 "proofState": 1,
 "goals":
 ["case h\nx y z : Nat\n⊢ (foo.match_1 (fun x => Prop) (x, y, z) fun x fst snd => x = x) = True"]}

{"proofStatus": "Incomplete: open goals remain",
 "proofState": 2,
 "goals": ["case h\nx y z : Nat\n⊢ (x = x) = True"]}

{"proofStatus": "Completed", "proofState": 3, "goals": []}

{"proofStatus": "Completed", "proofState": 4, "goals": []}

{"env": 1}

